edge-client/reset_factory_settings.c

```diff
@@ -132,6 +139,90 @@ EDGE_LOCAL void rfs_reset_factory_settings_request_cb(void *arg)
     free(arg);
 }
 
+#ifdef RFS_GPIO
+
+#ifdef RFS_GPIO_INVERTED
+#define RFS_GPIO_POLARITY (GPIOD_LINE_EVENT_FALLING_EDGE ^ GPIOD_LINE_EVENT_RISING_EDGE)
+#else
+#define RFS_GPIO_POLARITY 0
+#endif
+
+static void *rfs_gpio_thread(void *arg)
+{
+    struct gpiod_line *line = arg;
+    struct gpiod_line_event last_event = { 0 };
+    last_event.event_type = GPIOD_LINE_EVENT_FALLING_EDGE ^ RFS_GPIO_POLARITY;
+
+    for (;;) {
```

Comment:
Does this block everything else while it's looping?

Reply:
It's a whole dedicated thread for monitoring the GPIO line.

So this thread will spend all its life blocking in `gpiod_line_event_read`.

Reply:
Good!